[3.3.5] DB/Scripts: Migrate Raliq the Drunk and Sal'salabim to SAI

mysterious-static opened the pull request:
**Changes proposed:**

-  2 more SAI-scriptable NPCs to SAI, removing unneeded C++ scripts

**Target branch(es):**

- [X] 3.3.5

**Tests performed:** (Does it build, tested in-game, etc.)
Builds, tested

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] Refactor SQL after review

Commit:
raliq and sal'salabim to SAI

Commit:
Newline

Commit:
remove ScriptedGossip include

Commit:
typo

Commit:
no idea what I pasted earlier but it was clearly wrong

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,39 @@
+(18584, 0, 2, 3, 61, 0, 100, 0, 0, 0, 0, 0, 2, 90, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Gossip Option 0 Selected - Set Faction Demon'),
+(18584, 0, 3, 4, 61, 0, 100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Gossip Option 0 Selected - Say Line 0'),
+(18584, 0, 4, 0, 61, 0, 100, 0, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Gossip Option 0 Selected - Start Attacking'),
+(18584, 0, 5, 6, 32, 0, 100, 0, 0, 10000, 0, 0, 26, 10004, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Damaged Between 0-10000 - Quest Credit \'Patience and Understanding\''),
```
Instead of On Damaged Between 0-10000, set invincibility on gossip select and trigger the quest credit on 5% HP

velinath replied:
Oooh, thanks. That's a lot more elegant. I'll update this shortly.

Commit:
Update smart_script to use invincibility

I used 19% for invincibility pct because the smart_event for "health below %" reads as < and not <=. If they can both be set to 20% I can make that change, but since I'm not at a place I can test I figured I'd go better safe than sorry now and then change it later if I need to.

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,40 @@
+(14, 7725, 9435, 0, 0, 9, 0, 10004, 0, 0, 0, 0, 0, '', 'Show gossip menu 7725 text id 9440 if quest Patience and Understanding has been taken.'),
+(15, 7725, 0, 0, 0, 9, 0, 10004, 0, 0, 0, 0, 0, '', 'Show gossip menu 7725 option id 0 if quest Patience and Understanding has been taken.'),
+(22, 6, 18584, 0, 0, 32, 0, 16, 0, 0, 0, 0, 0, '', 'Creature SmartAI event 5 for Sal\'salabim will execute if invoker is player.'),
+(22, 6, 18584, 0, 0, 38, 1, 20, 4, 0, 0, 0, 0, '', 'Creature SmartAI event 5 for Sal\'salabim will execute if creature health percentage is equal or lower than 20% of max Health.');
```
This 2 SAI conditions are useless.

velinath replied:
It wouldn't still need the "if invoker is player" one, just updated to point to the proper event? It was in the C++ script so I pulled it over

Killyana replied:
The invoker is the one who used the gossip option it could be only a player with the quest taken.

Killyana replied:
Ah I see, conditions will change nothing in this case, you can change the credit target for all target in the threat list instead of the invoker, or store the target on gossip select and give the credit to the stored target

Commit:
remove conditions, per Killyana

Commit:
target threat list for quest credit

(as opposed to action_invoker)

Commit:
forgot to put semicolon after removing conds

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,38 @@
+(18584, 0, 3, 4, 61, 0, 100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Gossip Option 0 Selected - Say Line 0'),
+(18584, 0, 4, 5, 61, 0, 100, 0, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Gossip Option 0 Selected - Start Attacking'),
+(18584, 0, 5, 0, 61, 0, 100, 0, 0, 0, 0, 0, 42, 0, 19, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Gossip Option 0 Selected - Set Invincibility Hp 19%'),
+(18584, 0, 6, 7, 2, 0, 100, 0, 0, 20, 0, 0, 26, 10004, 0, 0, 0, 0, 0, 24, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Health Below 20% - Quest Credit \'Patience and Understanding\''),
```
Last changes: Action here must be 15

velinath replied:
this is as a result of using threat list as target?

Killyana replied:
No, you can also test with target 16, it will not complete the quest.

velinath replied:
Sounds good, I'll update. It was working with action_invoker but I imagine that's because it was only one target maybe?

velinath replied:
btw I very much appreciate the detailed review, it's helping me learn a lot about the sai system that wasn't clear from wiki and making it easier for me to understand what needs to happen in future scripts

Commit:
update to call_areaexploredoreventhappens

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,38 @@
+(18584, 0, 3, 4, 61, 0, 100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Gossip Option 0 Selected - Say Line 0'),
+(18584, 0, 4, 5, 61, 0, 100, 0, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Gossip Option 0 Selected - Start Attacking'),
+(18584, 0, 5, 0, 61, 0, 100, 0, 0, 0, 0, 0, 42, 0, 19, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Gossip Option 0 Selected - Set Invincibility Hp 19%'),
+(18584, 0, 6, 7, 2, 0, 100, 0, 0, 20, 0, 0, 15, 10004, 0, 0, 0, 0, 0, 24, 0, 0, 0, 0, 0, 0, 0, 'Sal\'salabim - On Health Below 20% - Quest Credit \'Patience and Understanding\''),
```
No repeat for the event_flag:
SmartAIMgr::LoadSmartAIFromDB: Entry 18584 SourceType 0, Event 6, Missing Repeat flag.

velinath replied:
sorry, i'm lost here based on the documentation. should be flag 1? maybe I'm too tired to understand right now, but just not quite getting it

Killyana replied:
Some events requires repeat delay or no repeat flags, almost all the event using the "update" (IC, OOC, HPC ect...)

Killyana committed:
Update 9999_99_99_99_world_335.sql

Aokromes committed:
Rename 9999_99_99_99_world_335.sql to 2017_10_21_03_world.sql
